src/main/java/core/basesyntax/service/RegistrationServiceImp.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+
+public class RegistrationServiceImp {
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImp(StorageDaoImpl storage) {
+        this.storageDao = storage;
+    }
+
+    public void register(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
```

Comment:
You need to check if the user's login is null before calling the get method on it. If it's null, it will throw a NullPointerException. Instead, you should throw an InvalidUserDataException with an informative message.